### ISSUE TITLE
Notice Requirements

### ISSUE THREAD

**Issue by heathermeeker** [kemitchell/api-copyleft-license#5]
"You must ensure that everyone who gets a copy of any part of this software from you, in source code or any other form, with or without changes, also gets the text of this license, including its developer and homepage lines."

A few thoughts, which you have probably considered already...

- Is the intention to lay aside notices for users who get no copy but use software via SaaS? 
- I prefer to allow short form notice requirements. I realize this license is a lot shorter than AGPL (and that is good), but the need to copy license text is the number-one complaint of my clients about open source licenses. In MPL2 I recall we allowed a short form notice only with a link to the license. Or maybe you are contemplating that a link is sufficient to get the license?
- Embedding dev information in licenses is a double edged sword. On one had it means no hunting for copyright notices to convey. On the other, it makes each instance of the license unique. Overall I actually prefer the embedded form, if the license is short.

**Comment by kemitchell** [comment]
@heathermeeker, thanks for these thoughts! I'll respond in order.

**Comment by kemitchell** [comment]
## SaaS Notice

@heathermeeker  and I have discussed "closing the ASP loophole" in notice provisions elsewhere. For the benefit of others:

- Typical open source notice conditions only require giving copies of license terms and copyright notices with copies of the software.
- Those typical conditions therefore do _not_ require giving copies of license terms and copyright notices to users of hosted software.
- That creates an "ASP/SaaS loophole", just like triggering GPLv2 copyleft on distribution created a loophole.
- Requiring notices as part of SaaS use would ensure open source contributors more public credit.
- While the legal mechanisms underlying typical open source warranty disclaimers and damages exclusions aren't clear, ensuring notice of those terms reach end users could help to protect open source contributors.

My intention in first draft here was _not_ to "close the ASP notice loophole", but instead to follow current orthodoxy. But that decision fell out of a desire to focus on the other ways this license is novel.

In other words, this license hasn't quite decided exactly what it wants to be when it grows up. I'm completely open to doing "SaaS notice" in this form. I suppose that should come down to whether potential licensors on these terms, as we anticipate them, will want it, net of anticipated effect on compliance hassles, and potentially adoption.

**Comment by kemitchell** [comment]
## Short-Form Notice Requirements

When it comes to notice with distribution, I'm not so sympathetic to calls for notice by URL or an abbreviated statement. `LICENSE` files and header comments are bits, just like software is bits. If you're sending 100K of software, an additional 4K of `LICENSE` is no burden.

It's often _more_ work to pull files and comments _out_ of a artifact that you received. Moreover, when there's some translation step from artifact received to copy distributed, such as compiling C code or bundling JavaScript, I'd rather address the "inconvenience" of preserving notices with software than legal terms. Tools do and should exist for compiling `NOTICES` files, for all kinds of open source packaging conventions. If we get to pick between solving a problem with software and solving it with legal, solve it with software.

That being said, if the license ends up closing the "SaaS notice loophole", I'll immediately become inclined to allow some form of short-form notice. In that case, the means of sharing the software and the means of sharing the notices is no longer the same.

**Comment by bhilburn** [comment]
@kemitchell - The biggest counter point I can think of to the first item, above: 

> I anticipate licensors under these terms will tend to be unitary IP owners.

Is that this might imply that a licensor must obtain CLAs from contributors, rather than enabling things like `inbound=outbound`.

I recognize that the license doesn't actually require that, but the structure of the dev information does clearly prefer a single copyright holder, as you said.

**Comment by kemitchell** [comment]
Under #40, there's no longer any fill-in-the-blank lines.
